File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var mergeNodes = function(head) {
    let result = new ListNode();
    let itterator = head.next;
    let resultItterator = result;
    let val = 0;

    while (itterator != null){
        if (itterator.val == 0){
            resultItterator.val = val;
            if (itterator.next != null){
                let temp = new ListNode();
                resultItterator.next = temp;
                resultItterator = resultItterator.next;
            }
            val = 0;
        }
        else{
            val += itterator.val;
        }
        itterator = itterator.next;
    }
    return result;
};

File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        result = ListNode()
        t = result
        temp = head.next
        curr = 0
        while temp:
            if temp.val == 0:
                result.val = curr
                curr = 0
                if temp.next:
                    temp2 = ListNode()
                    result.next = temp2
                    result = temp2
            else:
                curr += temp.val
            temp = temp.next
                
        return t
        